### DISCUSSION TITLE
Problème installation

### DISCUSSION THREAD

**Discussion by Alex968-fr** [dbarzin/mercator#780] [Q&A]
Bonjour,

j'essaie d'installer Mercator sur un conteneur Ubuntu, situer dans mon serveur Debian 11.
J'ai besoin de cette application pour l'entreprise dans laquelle je travail, pour effectuer la cartographie de tous les ystèmes.
J'ai suivi toutes les étapes d'installation, tout à l'air ok, mais quand je vais sur l'interface web, c'est incomplet et surtout non mis en forme...
Je ne m'y connais pas trop dans ce genre d'installation et j'avoue être un peu perdu.. 
Je vous joins une capture d'écran pour vous aider à visualiser le problème, si quelqun ici peut m'aider ce serais sympa :)

Merci d'avance !
![Mercator 1](https://github.com/user-attachments/assets/990f79f2-dce9-4aeb-b425-af81b1c579e1)
![Mercator 2](https://github.com/user-attachments/assets/dcb418fa-eba9-427b-8cfc-2a02a21d05c0)

**Comment by dbarzin** [comment]
As-tu bien suivi la procédure d'installation ?
As-tu rencontré des problèmes durant l'installation ?
As-tu configuré l'application pour fonctionner avec Apache ou l'as-tu lancée avec la commande `php artisan serve` ?

**Reply by Alex968-fr** [reply]
Bonjour,

merci de votre retour, oui j'ai bien suivi toute la procédure d'installation Ubuntu et je n'ai pas rencontrer de problèmes.
Je l'ai lancé avec la commande php artisan serve.

avez vous besoin de capture d'écran du répertoire mercator et de ce qu'il contient ?

**Reply by dbarzin** [reply]
C'est étrange, car la capture d'écran montre que c'est Apache qui tourne sur le port 8000. 
Qu'est-ce qui s'affiche lors du lancement de la commande `php artisan serve` ?

**Comment by Alex968-fr** [comment]
il affiche server running
![Mercator 3](https://github.com/user-attachments/assets/102f3a3d-3bf8-42f5-a227-0ed331e1ed1c)

**Reply by dbarzin** [reply]
et quand tu vas sur http://127.0.0.1:8000, tu as quoi ?

**Reply by Alex968-fr** [reply]
et bien j'ai la même chose que la première capture d'écran.
![Mercator 1](https://github.com/user-attachments/assets/4e2fac71-1d74-4105-af5c-e05666144b1e)

**Reply by dbarzin** [reply]
Est-ce qu'il y a une erreur dans `storage/logs/laravel.log` ?

**Reply by Alex968-fr** [reply]
je m'excuse je ne trouve pas le répertoire demander ... 
![Mercator 2](https://github.com/user-attachments/assets/890a381e-7bb6-4516-9d76-e6c440d1ffa2)
est-il censé être à un autre endroit ? j'ai également regarder dans var/log et je ne trouve pas laravel.log

**Reply by Alex968-fr** [reply]
j'ai pourtant bien lancé toutes les commandes y compris pour laravel 
![Mercator 3](https://github.com/user-attachments/assets/8e4bc0c9-40c0-4356-810d-2fcd06aa0b10)

**Comment by dbarzin** [comment]
Serait-il possible que le processus `'php artisan serve'` n'ait pas le droit d'écrire dans l'arborescence de fichiers sous `'/var/www/mercator' `?
Cela expliquerait l'absence de logs et la réponse étrange du serveur.

**Reply by Alex968-fr** [reply]
peut-être, bien que normalement il n'y ai pas de raison que ce processus n'ai pas les droits ? 

comment puis-je vérifier cela svp ?

**Comment by dbarzin** [comment]
C'est du Linux de base : `ls -lsa /var/www/mercator`.

**Reply by Alex968-fr** [reply]
je viens de verifier et artisan n'avait que les droits d'écriture propriétaire.
![Mercator 1](https://github.com/user-attachments/assets/dd4957a5-c40c-4935-8d3b-468ed1d97bb5)

 je lui ai donc accorder tous les droit avec chmod 7777 pour tester.
![Mercator 2](https://github.com/user-attachments/assets/7304057a-dd0b-4a3f-9c6e-f7199a9dfb2f)

ce ne change rien malheureusement même après un restart du conteneur.

**Comment by Alex968-fr** [comment]
Bonjour,

je reviens vers vous car j'ai résolu mon problème, c'était bien un problème avec apache2, il n'avait pas pris en compte la réécriture avec les commande suivante : 
sudo a2enmod rewrite
sudo a2dissite 000-default.conf
sudo a2ensite mercator.conf

ceci dit quand je me suis connecter avec les identifiant par défaut cela m'a afficher une autre erreur : 
500 | server error
cela apparait juste après m'être connecter, je n'ai donc pas accès au reste.

Merci d'avance.

**Reply by Alex968-fr** [reply]
Edit : après vérification des logs, cela venait de la base de donnée MySQL qui n'était pas démarrer

l'application fonctionne parfaitement maintenant. :)

merci de votre aide !

Bonne journée.
